### DISCUSSION TITLE
How to improve the convergence of CZM ?

### DISCUSSION THREAD

**Discussion by echo1115** [idaholab/moose#22683] [Q&A Modules: Solid mechanics] [answered]
Hi all,
When I simulating a thermal-mechanical case with CZM model, it became difficult to converge after softening onset (damage>0). According to https://mooseframework.inl.gov/source/materials/cohesive_zone_model/BiLinearMixedModeTraction.html#Camanho2002, 
It seems the parameter _viscosity_  can help, but I don't know its value range and how to set it appropriate according to specific case. 

Here is my `czm `setting:
```
  [czm]
    type = BiLinearMixedModeTraction
    boundary = 'Block5_matrix'
    penalty_stiffness = 1.0e15 
    GI_c = 25 
    GII_c = 25 
    normal_strength = 100e6
    shear_strength = 240e6
    eta = 1 
    viscosity = 1 
    mixed_mode_criterion = POWER_LAW #BK
  []
```

Maybe someone could give me suggestions. Thanks in advance.

**Comment by hugary1995** [comment, marked as answer]
Yeah, the convergence issue is common to all softening models. Typical ways of dealing with this include
1. alternating minimization.
2. lagging the state variables.
3. nonlinear preconditioner.

You can set up 1 using fixed point iterations.
Option 2 is recommended for CZM, where you would set `lag_mode_mixity = true` and `lag_displacement_jump = true`.
Option 3 is not yet available in MOOSE, but it is on the roadmap.

**Reply by echo1115** [reply]
@hugary1995 Thanks for your useful suggestions!
